### PR TITLE
HHH-13861: Expose the doWork() and doReturningWork() APIs on StatelessSession as well

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-13861

### CODE REVIEW

**Comment by dreab8** [comment]
Thanks @gbadner , I applied your suggestion

**Comment by Sanne** [comment]
testing...

**Comment by Sanne** [comment]
I think in this case we might not want to have those methods with a "default" implementation:

- we don't support custom implementation of `org.hibernate.StatelessSession` so adding the methods to the interface isn't going to break code.
- the defualt implementation isn't functional

File: hibernate-core/src/test/java/org/hibernate/test/stateless/StatelessDoWorkTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+					(connection) -> {
+						PreparedStatement preparedStatement = connection.prepareStatement(
+								"SELECT NAME FROM TEST_ENTITY WHERE ID = ?" );
+						preparedStatement.setInt( 1, PERSISTED_TEST_ENTITY_ID );
+						ResultSet resultSet = preparedStatement.executeQuery();
+						String name = null;
+						if ( resultSet.next() ) {
+							name = resultSet.getString( 1 );
+						}
+						return name;
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/test/stateless/StatelessDoWorkTest.java]: I know `PreparedStatement`, `ResultSet` will be closed automatically when `Connection` is closed, but it seems it is not difficult to make the code more robust as following
```
try ( PreparedStatement preparedStatement = connection.prepareStatement(
								"SELECT NAME FROM TEST_ENTITY WHERE ID = ?" ) ) {
       preparedStatement.setInt( 1, PERSISTED_TEST_ENTITY_ID );
						
      try (ResultSet resultSet = preparedStatement.executeQuery() ) {
          String name = null;
	  if ( resultSet.next() ) {
		name = resultSet.getString( 1 );
	  }
	  return name;
    }
}
```
ignore the comment if you think it is too trivial. TBH, I hesitated a lot to comment.

File: hibernate-core/src/test/java/org/hibernate/test/stateless/StatelessDoWorkTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	}
+
+	@Test
+	public void testDoWork() {
+		try (StatelessSession statelessSession = sessionFactory().openStatelessSession()) {
+			statelessSession.doWork(
+					(connection) -> {
+						PreparedStatement preparedStatement = connection.prepareStatement(
+								"DELETE FROM TEST_ENTITY " );
+						preparedStatement.execute();
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/test/stateless/StatelessDoWorkTest.java]: Ignore this comment if you think it is too trivial:
```
try ( PreparedStatement preparedStatement = connection.prepareStatement(
								"DELETE FROM TEST_ENTITY " ) ) {
       preparedStatement.execute();
}
```

**Comment by dreab8** [comment]
@NathanQingyangXu I applied you suggestion.

@Sanne I changed the 2 default methods as we discussed, now they throw an `UnsupportedOperationException` exception.

Thanks for the reviews.

**Comment by Sanne** [comment]
merged. thanks all!

**Pull request closed without merging**